### PR TITLE
Converted more molnet loaders to new API

### PR DESCRIPTION
**Pull request by peastman**
This includes all the materials datasets, as well as PCBA.  @ncfrey if you could look this over to make sure I converted them correctly, that would be great!

I removed the test cases for the materials datasets, because they didn't actually load the real datasets, and relied on features that don't exist in the new API.

For the moment I removed the pcba_146 and pcba_2475 loaders, since the data files no longer exist on the server.  If we figure out what happened to them and restore them, we can add those functions back.

### CODE REVIEW

File: deepchem/trans/transformers.py [hunk 1]
```diff
@@ -972,6 +972,8 @@ def transform_array(self, X, y, w, ids):
       Transformed array of ids
     """
     w_balanced = np.zeros_like(w)
+    if len(y.shape) == 1 and len(w.shape) == 2 and w.shape[1] == 1:
```
Comment (rbharath) [inline deepchem/trans/transformers.py:975]: It might be good to add a unit test to document the fix this is making. Shape bugs are nasty so getting a nice set of unit tests to catch corner cases could help a lot

**Review by rbharath — COMMENTED** [review]
Did a quick pass, but will defer to @ncfrey for more detailed review :)

File: deepchem/molnet/load_function/material_datasets/load_bandgap.py [hunk 2]
```diff
@@ -112,91 +93,13 @@ def load_bandgap(
   Examples
   --------
   >> import deepchem as dc
-  >> tasks, datasets, transformers = dc.molnet.load_bandgap(reload=False)
+  >> tasks, datasets, transformers = dc.molnet.load_bandgap()
   >> train_dataset, val_dataset, test_dataset = datasets
   >> n_tasks = len(tasks)
   >> n_features = train_dataset.get_data_shape()[0]
   >> model = dc.models.MultitaskRegressor(n_tasks, n_features)
```
Comment (ncfrey) [inline deepchem/molnet/load_function/material_datasets/load_bandgap.py:100]: Is it possible to use `>>>` now so that these examples render correctly in the docs and are checked with doctest?

**Review by ncfrey — COMMENTED** [review]
This looks great! It really cuts down on code duplication and is much cleaner. I think the defaults for the materials datasets look good. 

It's outside of the scope of this PR and might have been mentioned before, but it could be helpful to have some minimal testing template that mocks the downloading and featurization of datasets.

File: deepchem/molnet/load_function/material_datasets/load_bandgap.py (hunk 2 above)
Comment (peastman) [reply deepchem/molnet/load_function/material_datasets/load_bandgap.py:100]: I don't think we can have doctest running these examples, because that would make it way too slow.

Comment (rbharath) [reply deepchem/molnet/load_function/material_datasets/load_bandgap.py:100]: @nd-02110114 Suggested a trick for the rendering:

```
>>> ...
>> tasks, datasets, transformers = dc.molnet.load_bandgap()
>> ...
```

That is, you have the first line with a `>>>` to render it correctly, but you don't actually run the code. I think there are a few examples of this elsewhere in DeepChem

Comment (peastman) [reply deepchem/molnet/load_function/material_datasets/load_bandgap.py:100]: I'll give it a try!

Comment (peastman) [reply deepchem/molnet/load_function/material_datasets/load_bandgap.py:101]: Fixed!

File: deepchem/trans/transformers.py (hunk 1 above)
Comment (peastman) [reply deepchem/trans/transformers.py:975]: Added!

**Review by rbharath — COMMENTED** [review]
@peastman Feel free to merge this in once ready :). I think all outstanding comments have been addressed
